azurelinuxagent/ga/agent_update.py

```diff
@@ -170,15 +172,20 @@ def __get_all_agents_on_disk():
         path = os.path.join(conf.get_lib_dir(), "{0}-*".format(AGENT_NAME))
         return [GuestAgent.from_installed_agent(path=agent_dir) for agent_dir in glob.iglob(path) if os.path.isdir(agent_dir)]
 
-    @staticmethod
-    def __log_event(level, msg_, success_=True):
-        if level == LogLevel.WARNING:
-            logger.warn(msg_)
-        elif level == LogLevel.ERROR:
-            logger.error(msg_)
-        elif level == LogLevel.INFO:
+    def __log_event(self, level, msg_, success_=True):
+        if level == LogLevel.INFO:
             logger.info(msg_)
-        add_event(op=WALAEventOperation.AgentUpgrade, is_success=success_, message=msg_, log_event=False)
+            add_event(op=WALAEventOperation.AgentUpgrade, is_success=success_, message=msg_, log_event=False)
+        else:
+            msg_ += "[NOTE: Will not log the same error for the next 6 hours]"
+            if msg_ != self._last_warning or self._last_warning_time == datetime.datetime.min or datetime.datetime.now() >= self._last_warning_time + datetime.timedelta(hours=6):
```

Comment:
does the message include any correlation id or anything that may be different in each invocation and make msg_ != self._last_warning be False?

Reply:
Great point, few messages include incarnation and may change on every goal state. Let me think about addressing that..

Reply:
how often could this be logged if we didn't suppress?

Reply:
On every iteration, we check agent updates, so the frequency is 6 secs. But we have additional check of one hour window to allow updates. Before we compute this window there could be failures like missing agent family and we end up logging this error msg until any GS includes family section

Reply:
updated the logic and compare the substring if incarnation included in msg